### DISCUSSION TITLE
SQLite Support

### DISCUSSION THREAD

**Discussion by biapar** [keystonejs/keystone#5334] [Questions] [answered]
Will be the SQLite support?

**Comment by timleslie** [comment, marked as answer]
@biapar We are currently working on adding SQLite support via the Prisma adapter. See https://github.com/keystonejs/keystone/pull/3946 for details.

**Comment by bladey** [comment]
Hi @biapar, we now have SQLite support, let us know if you have any questions!

**Comment by biapar** [comment]
Good. So I can give a try.

**Comment by NoTuxNoBux** [comment]
I found this thread and have a question: I have Jest tests set up by following the documentation and using `setupTestEnv`, and now switched from using a PostgreSQL test-database to using an SQLite database file, which seems to be noticeably faster when needing to up the schema already. It would likely be even faster if I could use an in-memory database instead, which avoids disk I/O entirely. [SQLite supports this](https://sqlite.org/inmemorydb.html) by passing `file::memory:` as URL, but I then get Keystone errors that my schema doesn't exist anymore (e.g. `The table main.User does not exist in the current database.`) - it works fine when using a file.

I'm guessing that Keystone's `setupTestEnv` is making its own connection to the database URL to up the schema, and then disconnecting, after which I have to do `await testEnv.connect();`, which, due to it being an in-memory database, is discarding the entire database and then creating a new one.

I'm guessing this is currently not supported, but I thought I'd ask or mention it anyway :slightly_smiling_face: .

SQLite in-memory databases facilitate natural partitioning between Jest tests - as the documentation recommends not sharing a database between tests - as well as parallelizing them so they don't have to run in-band, as they all have individual state anyway, at a very low performance cost.

**Comment by biapar** [comment]
How to use SQLite? Could you update the install documentation?

**Reply by bladey** [reply]
Hi @biapar, this is now released as per the other thread - https://github.com/keystonejs/create-keystone-app/pull/202
